src/components/Header/UserMenu.tsx

```diff
@@ -0,0 +1,78 @@
+import { Fragment } from 'react';
+import { HiUser } from 'react-icons/hi';
+import { Menu, Transition } from '@headlessui/react';
+import { type User } from '@supabase/gotrue-js/src/lib/types';
+import cn from 'classnames';
+
+import { useSignOut } from './hooks';
+
+export type UserMenuPros = {
+	user: User;
+	className?: string;
+};
+
+type MenuLink = {
+	label: string;
+	to: string;
+};
+
+const itemDefaultClassName = 'block px-4 py-2 text-sm text-gray-700';
+const menuLinks: MenuLink[] = [
```

Comment:
I suggest we use the generic way for better readability
```suggestion
const menuLinks: Array<MenuLink> = [
```

Reply:
We could set the rule if you want: https://typescript-eslint.io/rules/array-type/
```ts
"@typescript-eslint/array-type": ["error", { "default": "generic" }]
```

Reply:
@benny123tw that would be a nice addition. Can you create a PR for it?

Reply:
@iShibi sure. I'll create a PR.